subsys/caf/modules/buttons.c

```diff
@@ -179,8 +206,19 @@ static int suspend(void)
 		break;
 
 	case STATE_ACTIVE:
-		state = STATE_IDLE;
-		err = callback_ctrl(true);
+		/* Disable callbacks (and cancel the work) to ensure it will not be scheduled by an
+		 * invalid button in the idle state.
+		 */
+		err = callback_ctrl(MASK_NONE);
+
+		if (!err) {
+			/* Setup callbacks and columns for the idle state. */
+			state = STATE_IDLE;
+			err = set_cols(mask_wakeup(col, ARRAY_SIZE(col)), true);
+			if (!err) {
+				err = callback_ctrl(mask_wakeup(row, ARRAY_SIZE(row)));
+			}
```

Comment:
Here it should be actually fine - callbacks are disabled so no spurious action should be triggered here (actually it would be even better than enabling back the callbacks). The set columns state should not lead to any side effects in behaviour too.